api/main.py

```diff
@@ -259,6 +259,199 @@ def settle_btc_payments():
+
+                            else:
+                                app.logger.info(f"Payment for order.id={order.id} not received yet.")
+
+                        # OUTGOING PAYMENTS
+                        outgoing_payments_sent = True;
+
+                        if incoming_payment_received:
+                            payout_information = get_payout_information(order.seller_id)
+                            app.logger.info(f"Payout information - payout_information = {payout_information}")
+
+                            if not payout_information:
+                                app.logger.error(f"ERROR: There is no payout information for order.id={order.id}, order.seller_id={order.seller_id}  !!!!!")
+                                outgoing_payments_sent = False
+                            else:
+                                for payout in payout_information:
+                                    payout_ln_address = payout['ln_address']
+                                    payout_percent = payout['percent']
+                                    payout_amount = order.total * payout_percent / 100
+                                    payout_amount = round(payout_amount)
+
+                                    if ln_payment_logs_util.check_outgoing_payment(order.id, invoice.id, payout_ln_address, payout_amount):
+                                        app.logger.info(f"Payment for order id={order.id}, ln_address={payout_ln_address}, amount={payout_amount} WAS already paid.")
+                                    else:
+                                        app.logger.info(f"Paying for order id={order.id}, ln_address={payout_ln_address}, amount={payout_amount}...")
+                                        
+                                        if not invoice_util.pay_to_ln_address(payout_ln_address, payout_amount, 'Payment from order #{order.uuid}'):
+                                            outgoing_payments_sent = False
+                                            app.logger.error(f"ERROR: Couldn't made some outgoing payment!!! payout_ln_address={payout_ln_address}, payout_amount={payout_amount}  !!!!!")
+                                        else:
+                                            ln_payment_logs_util.add_outgoing_payment(order.id, invoice.id, payout_ln_address, payout_amount)
+
+                        if incoming_payment_received and outgoing_payments_sent:
+                            app.logger.info(f"    ********* EVERYTHING DONE, SO MARKING THIS PAYMENT AS PAID *********")
+                            order.paid_at = datetime.utcnow()
+                            db.session.commit()
+
+            except:
+                app.logger.exception("Error while getting information about Lightning Network payments.")
+                #db.session.rollback()
+
+        else:
+            app.logger.info(f"There aren't active orders with Lightning Network pending. Sleeping for a while.")
+
```

Comment:
There is some additional logic in settle-btc-payments that also needs to happen here, so it should somehow be extracted and called in both places:

* when the order is about purchasing a badge and the payment was completed, we need to award the badge and also look for pending bids and accept them...

Reply:
Also, settle-btc-payments seems to have the ability to "expire" orders, but this lightning processor not... It probably works, but it is counter-intuitive and risky. How if we move the expiring of orders somewhere else? Can we maybe have a new process that just looks at unpaid orders and expires them? It seems like that part should be decoupled from BTC now that we also have lightning...